### DISCUSSION TITLE
Export textured terrain from Godot.net to godot for html export

### DISCUSSION THREAD

**Discussion by zengels** [spimort/TerraBrush#174] [General]
First, I want to thank you for your remarkable work on Terrabrush. It’s an outstanding tool with impressive features for modeling and texturing terrains directly within Godot.

I have a technical question. Since Terrabrush operates exclusively with Godot.net, I would like to know if it’s possible to export a fully modeled and textured terrain created using Terrabrush in Godot.net, and then import it into Godot Core Classic as a standalone, fully textured 3D mesh or in another compatible format. The goal is to enable a web export, which is unfortunately not supported by Godot.net.

Thank you very much for your work and your assistance. Please let me know if you need further details.

Best regards,

**Comment by spimort** [comment]
Hey @zengels !

I'm glad you enjoy it! 🙂

Right now there is no way to export the actual mesh behind the terrain.
The reason for it is that you're gonna miss all the LOD features from the terrain since it would be a static mesh.

If you really want to do this, without the LOD features, you could export the images and use them to generate a terrain with a shader on a mesh instance. Obviously that would require some work to be done on your side.

The other option would be to export it and reimport it using other terrain tools (if the tool support importing).

I hope this helps 🙂

**Reply by zengels** [reply]
Hello! Thank you for your response! I've spent so much time trying to find an accessible solution to create a terrain using Godot, while ensuring it works seamlessly for web export. Finally, I discovered Godot Pipeline, which seems to be the right approach for me—creating terrains in Blender and syncing them with Godot.

Although bugs occasionally arise from one software or the other, I've been able to find workarounds so far, especially with Godot Pipeline, which has helped address several unintended behaviors. I never imagined that creating a simple terrain could turn into such an adventure!

Currently, I don’t know of another tool that allows exporting terrains directly from your excellent plugin, but I truly hope it becomes possible someday. As much as I'd like to explore that further, I already have a lot on my plate with my own project, leaving little time to focus on terrain creation.

Thank you again for this fantastic plugin, and I wish you all the best with your own projects!

**Reply by spimort** [reply]
That's perfect, as long as you have a solution for your project, that's the only thing that matters!

I'll close this discussion here since you've found a solution, feel free to open a new one if needed 🙂
